### PR TITLE
feat: aws-kms submodule test

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

I added a test that checks if the resources of the encryption-at-rest-aws-kms submodule are created and works correctly.

### CODE REVIEW

File: modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+run "enable_encryption_at_rest_aws_kms_key" {
+  command = apply
+
+  module {
+    source = "./"
+  }
+
+  variables {
+    project_id           = run.create_project.project_id
+    aws_kms_key_arn      = "arn:aws:kms:us-east-2:358363220050:key/b6a1d91b-59ad-4cb8-9d15-64175ab4791b"
```
Comment (EspenAlbert) [inline modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl]: can we use an env var instead of hard-coding?

Comment (rubenVB01) [reply modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl]: I thought the env variables are for specific attributes (like the provider's parameters, etc) is it possible to do it with this?

Comment (EspenAlbert) [reply modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl]: I think you can use a variable named: `TF_VAR_aws_kms_key_arn`

**Review by EspenAlbert — APPROVED** [review]
LGTM. Can you add a run for testing the validation of kms_key_region too maybe?

File: modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl (hunk 1 above)
Comment (rubenVB01) [reply modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl]: done, it works :)

File: modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  variables {
+    project_id           = run.create_project.project_id
+    iam_role_name        = "mongodb-atlas-test-acc-tf-${run.create_project.project_id}"
+    iam_role_policy_name = "mongodb-atlas-test-acc-tf-${run.create_project.project_id}"
+    kms_key_region       = "us_east_2"
+  }
+
+  expect_failures = [
```
Comment (rubenVB01) [inline modules/aws-kms/tests/push_based_log_export_create_bucket_default_name.tftest.hcl:68]: this parameter in the test means that the test is going to pass iff there is a failure during the execution

**Pull request merged**